lib/meadow/pipeline/actions/create_transcode_job.ex

```diff
@@ -0,0 +1,76 @@
+defmodule Meadow.Pipeline.Actions.CreateTranscodeJob do
+  @moduledoc """
+  Action to create an AWS Elemental MediaConvert Job for an audio or video file
+  """
+
+  @mediaconvert_client Application.compile_env(:meadow, :mediaconvert_client)
+
+  alias Meadow.Data.{ActionStates, FileSets}
+  alias MediaConvert.{AudioTemplate, VideoTemplate}
+  alias Sequins.Pipeline.Action
+
+  use Action
+  use Meadow.Pipeline.Actions.Common
+
+  require Logger
+
+  @actiondoc "Create an AWS Elemental MediaConvert Job for an audio or video FileSet"
+
+  defp already_complete?(file_set, _) do
+    ActionStates.ok?(file_set.id, __MODULE__)
+  end
+
+  defp process(file_set, attributes, _false) do
+    ActionStates.set_state!(file_set, __MODULE__, "started")
+
+    source = file_set.core_metadata.location
+    destination = FileSets.streaming_uri_for(file_set)
+    mime_type = file_set.core_metadata.mime_type
+    user_metadata = Map.put(attributes, :file_set_id, file_set.id)
+
+    if supported_mime_type?(mime_type) do
```

Comment:
MIME type checking is good for this initial round, but once the new dispatcher is in place, it should be removed and unsupported files should error out via pattern matching (the same way unsupported sources error out down on line 65).